spec/policies/subject_rule_effect_policy_spec.rb

```diff
@@ -0,0 +1,119 @@
+  let(:expired_credential) { fake_credential expired: true }
+  let(:admin_credential) { fake_credential admin: true }
+
+  let(:workflow) { create :workflow }
+  let(:rule) { create :subject_rule, workflow: workflow }
+  let(:effect) do
+    create(
+      :subject_rule_effect,
+      action: 'retire_subject',
+      config: { foo: 'bar' },
+      subject_rule: rule
+    )
+  end
+
+  before { effect }
+
+  permissions '.scope' do
+    it 'returns no records when not logged in' do
+      expect(records_for(not_logged_in_credential)).to match_array(SubjectRuleEffect.none)
+    end
+
+    it 'returns no records when token has expired' do
+      expect(records_for(expired_credential)).to match_array(SubjectRuleEffect.none)
+    end
+
+    it 'returns all workflow subject rule effects for an admin' do
+      expect(records_for(admin_credential)).to match_array(SubjectRuleEffect.where(id: effect.id))
+    end
+
+    it 'returns no workflow subject rule effects when not a collaborator on any project' do
+      credential = fake_credential project_ids: []
+      expect(records_for(credential)).to match_array(SubjectRuleEffect.none)
+    end
+
+    it 'returns the workflow subject rule effects the user is a collaborator on' do
+      credential = fake_credential(project_ids: [workflow.project_id])
+      expect(records_for(credential)).to match_array(SubjectRuleEffect.where(id: effect.id))
+    end
+  end
+
+  permissions :create? do
+    it 'denies access when not logged in' do
+      expect(subject).not_to permit(not_logged_in_credential, workflow)
+    end
+
+    it 'denies access when token has expired' do
+      expect(subject).not_to permit(expired_credential, workflow)
+    end
+
+    # temp fix for to stop non-admin users modify the rule effects
+    it 'denies access to all user that are collaborators on the project' do
+      credential = fake_credential(project_ids: [workflow.project_id])
+      expect(subject).not_to permit(credential, workflow)
+    end
+
+    it 'grants access to an admin' do
+      expect(subject).to permit(admin_credential, workflow)
+    end
+  end
+
+  permissions :create?, :update? do
+    it 'denies access when not logged in' do
+      expect(subject).not_to permit(not_logged_in_credential, effect)
+    end
+
+    it 'denies access when token has expired' do
+      expect(subject).not_to permit(expired_credential, effect)
+    end
+
+    # temp fix for to stop non-admin users modify the rule effects
+    it 'denies access to all user that are collaborators on the project' do
+      credential = fake_credential(project_ids: [workflow.project_id])
+      expect(subject).not_to permit(credential, effect)
+    end
+
+    it 'grants access to an admin' do
+      expect(subject).to permit(admin_credential, effect)
+    end
+  end
+
+  permissions :index? do
+    it 'denies access to non-collaborators on the project' do
+      credential = fake_credential(project_ids: [workflow.project_id + 1])
```

Comment:
Is this +1  just to have a non-empty project_ids array? Why?

Reply:
it's not an empty array, rather an id that doesn't match the credentials project_ids, empty array would also work but this tests the implementation a little more

Reply:
so...yes. :) cool, makes sense.